app/kafka/manager/ClusterManagerActor.scala

```diff
@@ -242,6 +245,8 @@ class ClusterManagerActor(cmConfig: ClusterManagerActorConfig)
                 throw new IllegalArgumentException(s"No generated assignment found for topic ${ti.topic}")
               } { childData =>
                 val assignments = deserializeAssignments(childData.getData)
+                // check if any nonexistent broker got selected for reassignment
+                require(isValidAssignment(bl, assignments), "The assignments contain nonexistent broker(s)")
```

Comment:
This is a good check.  Can we surface the brokers which are nonexistent in the error such that it goes back through to the UI.